File: app/src/main/java/com/paxus/pay/poslinkui/demo/utils/Logger.java
package com.paxus.pay.poslinkui.demo.utils;

import android.content.Intent;

import androidx.annotation.NonNull;
import androidx.annotation.Nullable;

import java.util.Arrays;

/**
 * Logger, but more pretty, simple and powerful
 */
public final class Logger {

    private Logger() {
        //no instance
    }

    public static void d(@NonNull String message, @Nullable Object... args) {
        com.orhanobut.logger.Logger.d(message, args);
    }

    public static void d(@Nullable Object object) {
        com.orhanobut.logger.Logger.d(object);
    }

    public static void e(@NonNull String message, @Nullable Object... args) {
        com.orhanobut.logger.Logger.e(message, args);
    }

    public static void e(@Nullable Throwable throwable) {
        e(throwable, throwable == null || throwable.getMessage() == null ? "" : throwable.getMessage());
    }

    public static void e(@Nullable Throwable throwable, @NonNull String message, @Nullable Object... args) {
        com.orhanobut.logger.Logger.e(throwable, message, args);
    }

    public static void i(@NonNull String message, @Nullable Object... args) {
        com.orhanobut.logger.Logger.i(message, args);
    }

    public static void v(@NonNull String message, @Nullable Object... args) {
        com.orhanobut.logger.Logger.v(message, args);
    }

    public static void w(@NonNull String message, @Nullable Object... args) {
        com.orhanobut.logger.Logger.w(message, args);
    }

    /**
     * Tip: Use this for exceptional situations to log
     * ie: Unexpected errors etc
     */
    public static void wtf(@NonNull String message, @Nullable Object... args) {
        com.orhanobut.logger.Logger.wtf(message, args);
    }

    /**
     * Formats the given json content and print it
     */
    public static void json(@Nullable String json) {
        com.orhanobut.logger.Logger.json(json);
    }

    /**
     * Formats the given xml content and print it
     */
    public static void xml(@Nullable String xml) {
        com.orhanobut.logger.Logger.d(xml);
    }


    public static void intent(Intent intent) {
        StringBuilder intentBuilder = new StringBuilder();
        if(intent.getExtras() != null){
            for(String key : intent.getExtras().keySet()){
                intentBuilder.append(key).append(": ");
                boolean isArray = intent.getExtras().get(key) != null && intent.getExtras().get(key).getClass().isArray();
                intentBuilder.append(isArray ? stringifyArray(intent.getExtras().get(key)) : intent.getExtras().get(key));
                intentBuilder.append(System.lineSeparator());
            }
        }
        Logger.i(intentBuilder.toString());
    }

    private static String stringifyArray(Object array){
        switch (array.getClass().getSimpleName()){
            case "short[]":
                return Arrays.toString((short[])array);
            case "Object[]":
            case "String[]":
                return Arrays.toString((String[])array);
            default:
                return "UNABLE TO LOG";
        }
    }

}

File: app/src/main/java/com/paxus/pay/poslinkui/demo/utils/Logger.kt
package com.paxus.pay.poslinkui.demo.utils

import android.content.Intent
import com.orhanobut.logger.Logger as OrhanLogger

/**
 * Logger, but more pretty, simple and powerful
 */
object Logger {
    fun d(message: String, vararg args: Any?) {
        OrhanLogger.d(message, *args)
    }

    fun d(`object`: Any?) {
        OrhanLogger.d(`object`)
    }

    fun e(message: String, vararg args: Any?) {
        OrhanLogger.e(message, *args)
    }

    fun e(throwable: Throwable?) {
        val msg = throwable?.message.orEmpty()
        OrhanLogger.e(throwable, msg)
    }

    fun e(throwable: Throwable?, message: String, vararg args: Any?) {
        OrhanLogger.e(throwable, message, *args)
    }

    fun i(message: String, vararg args: Any?) {
        OrhanLogger.i(message, *args)
    }

    fun v(message: String, vararg args: Any?) {
        OrhanLogger.v(message, *args)
    }

    fun w(message: String, vararg args: Any?) {
        OrhanLogger.w(message, *args)
    }

    /**
     * Tip: Use this for exceptional situations to log
     * ie: Unexpected errors etc
     */
    fun wtf(message: String, vararg args: Any?) {
        OrhanLogger.wtf(message, *args)
    }

    /**
     * Formats the given json content and print it
     */
    fun json(json: String?) {
        OrhanLogger.json(json)
    }

    /**
     * Formats the given xml content and print it
     */
    fun xml(xml: String?) {
        OrhanLogger.d(xml)
    }


    fun intent(intent: Intent, vararg headings: String?) {
        val intentBuilder = StringBuilder()

        for (heading in headings) {
            intentBuilder.append(heading).append(System.lineSeparator())
        }

        intentBuilder.append("Intent: ").append(intent.action).append(System.lineSeparator())
        val extras = intent.extras
        if (extras != null) {
            for (key in extras.keySet()) {
                intentBuilder.append(key).append(": ")
                val raw = extras.get(key)
                if (raw == null) {
                    intentBuilder.append("null").append(System.lineSeparator())
                    continue
                }
                val isArray = raw.javaClass.isArray
                intentBuilder.append(
                    if (isArray) stringifyArray(raw) else raw
                )
                intentBuilder.append(System.lineSeparator())
            }
        }
        i(intentBuilder.toString())
    }

    private fun stringifyArray(array: Any): String {
        val kind = array.javaClass.simpleName
        return when (kind) {
            "short[]" -> (array as ShortArray).contentToString()
            "Object[]", "String[]" -> (array as Array<String?>).contentToString()
            else -> "UNABLE TO LOG"
        }
    }
}